### PR TITLE
DRILL-4410: ListVector should initialize bits in allocateNew

### PR DESCRIPTION
**Pull request by minji-kim**
ListVector does not initialize bits, but various ValueVectors depend on bits being initialized.  ONe of the side effects of not having bits set is that for each batch, the ValueVector will reAlloc() with 2x the previous buffer size (regardless of whether it needs it or not).  If there are enough batches, this results in reAlloc() request of size > Integer.MAX, which triggers out-of-memory/max-buffer exception.

### CODE REVIEW

**Comment by jaltekruse** [comment]
Can you generate these files in the tests rather than check them in?

**Comment by minji-kim** [comment]
That's a good point.  Just uploaded small patch to generate it.

**Comment by jaltekruse** [comment]
Could you also add result verification? While there are some older tests that just run queries to verify that errors that previously occurred are gone, we have been enforcing that new tests since the test builder was added verify their results. You can use the test builder to add records in a loop to the expected result set.

**Comment by minji-kim** [comment]
Added the checks in the test.

File: exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java [hunk 1]
```diff
@@ -241,4 +252,49 @@ public void testRepeatedJson() throws Exception {
             .go();
   }
 
+  @Test  // DRILL-4410
+  // ListVector allocation
+  public void test_array() throws Exception{
+
+    long numRecords = 100000;
+    String file1 = "/tmp/" + TestComplexTypeReader.class.getName() + "arrays1.json";
```
Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: Will this work correctly on windows environment?

Comment (minji-kim) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: ParquetRecordReaderTest also uses "/tmp", so I think this should also work.

Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: Seems ParquetRecordReaderTest is ignored?

[1] https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/ParquetRecordReaderTest.java#L84

Comment (adeneche) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: an alternative is to use BaseTestQuery.getTempDir("ComplexTypeWriter")

Comment (minji-kim) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: I think these tests all use /tmp. 

https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/impersonation/TestImpersonationMetadata.java#L64

https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/TestDropTable.java#L166

https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestWriter.java#L60

Comment (vkorukanti) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestComplexTypeReader.java]: TestImpersonationMetadata refers to a directory on HDFS which should be fine as the paths are unix style.

**Comment by minji-kim** [comment]
I made a change in the test to use the temporary directory, since that seems to be questionable.  Also I added another test in TestValueVector.  Both tests fail without the change in ListVector.

**Comment by jaltekruse** [comment]
+1 @hsuanyi Does this second test look good to you? I will test this on windows before merging, but I think it should be fine. Will do a small refactoring to use the generated dfs_test tmp space instead if it's an issue.

**Pull request closed without merging**